lambdas/helpers/dynamoDBHelper.js

```diff
@@ -2,40 +2,71 @@
 const AWS = require('aws-sdk');
 const log = require('./logger');
 
+const appConfig = {
+  test: {
+    domain: 'sites-test.federalist.18f.gov',
+    tableName: 'federalist-proxy-test',
+    siteKey: 'id',
+  },
+  staging: {
+    domain: 'sites-staging.federalist.18f.gov',
+    tableName: 'federalist-proxy-staging',
+    siteKey: 'id',
+  },
+  prod: {
+    domain: 'sites-prod.federalist.18f.gov',
+    tableName: 'federalist-proxy-prod',
+    siteKey: 'id',
+  },
+};
+
 const getSiteConfig = async (params) => {
   const docClient = new AWS.DynamoDB.DocumentClient({
     httpOptions: { connectTimeout: 120000, timeout: 120000 },
   });
-  return docClient.query(params)
+  return docClient.get(params)
     .promise()
     .catch((err) => {
       throw new Error(`Unable to query. Error: ${JSON.stringify(err, null, 2)}`);
     })
-    .then(({ Count, Items }) => {
-      if (Count > 0) {
-        const [item] = Items;
-        log(`\nQuery succeeded: item found @id:${JSON.stringify(item.id)}\n`);
-        return item.settings;
+    .then(({ Item }) => {
+      if (Item) {
+        log(`\nQuery succeeded: item found @id:${JSON.stringify(Item.id)}\n`);
+        return Item.settings;
       }
       log('\nQuery succeeded: no results found!!\n');
       return undefined;
     });
 };
 
-const getSiteQueryParams = (tableName, siteKey, siteKeyValue) => {
-  const expressionAttributeNames = {
-    [`#${siteKey}`]: siteKey,
-  };
+const validateProxyFunctionName = functionName => /^us-east-1:federalist-proxy-(prod|staging|test)-(viewer|origin)-(request|response):\d+$/
+  .exec(functionName);
 
-  const expressionAttributeValues = {
-    [`:${siteKey}`]: siteKeyValue,
-  };
+const getAppConfig = (functionName) => {
+  if (validateProxyFunctionName(functionName)) {
+    const appEnv = /prod|staging|test/.exec(functionName);
+    if (appEnv && appEnv.length) {
+      return appConfig[appEnv[0]];
+    }
+  }
+  throw new Error(`Unable to find appConfig for @function: ${functionName}`);
+};
+
```

Comment:
Why do we need to test the regex twice? Can't we just:
```
const functionRE =  /^us-east-1:federalist-proxy-(prod|staging|test)-(viewer|origin)-(request|response):\d+$/;

const getAppConfig = (functionName) => {
  const match = functionRE.exec(functionName);
  if (match === null) { // or `if (!match)`
    throw new Error(`Unable to find appConfig for @function: ${functionName}`);
  }
  return appConfig[appEnv[1]];
};
```

Reply:
@davemcorwin 
in my code, the first check is to validate the function and the second gets the appEnv
With your suggestion how are you defining appEnv ?

Reply:
Both are checking the regex using `.exec()`, if the regex doesn't match, it will return null.